android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java

```diff
@@ -304,28 +304,26 @@ private boolean isDeviceVersionSupported() {
 
   public static abstract class Stub {
 
-    public static AppcoinsBilling asInterface(IBinder service, String componentName) {
-      if (!WalletUtils.hasWalletInstalled()) {
+    public static AppcoinsBilling asInterface(IBinder service) {
+      if (WalletBinderUtil.getBindType() == BindType.WALLET_NOT_INSTALLED
+          || !WalletUtils.hasWalletInstalled()) {
         return AppcoinsBillingStubHelper.getInstance();
       } else {
         SharedPreferencesRepository sharedPreferencesRepository =
             new SharedPreferencesRepository(WalletUtils.getContext(),
                 SharedPreferencesRepository.TTL_IN_SECONDS);
         AppcoinsBilling appcoinsBilling;
-        if (UriCommunicationAppcoinsBilling.class.getSimpleName()
-            .equals(componentName)) {
-          SyncIpcMessageRequester messageRequester =
-              MessageRequesterFactory.create(WalletUtils.getContext(),
-                  BuildConfig.BDS_WALLET_PACKAGE_NAME,
-                  "appcoins://billing/communication/processor/1",
-                  "appcoins://billing/communication/requester/1",
-                  MESSAGE_RESPONSE_WAIT_TIMEOUT_IN_MILLIS);
+        if (WalletBinderUtil.getBindType() == BindType.URI_CONNECTION) {
+          SyncIpcMessageRequester messageRequester = MessageRequesterFactory.create(
+              new LifecycleActivityProvider(WalletUtils.getContext()),
+              BuildConfig.BDS_WALLET_PACKAGE_NAME, "appcoins://billing/communication/processor/1",
+              "appcoins://billing/communication/requester/1", BdsService.TIME_OUT_IN_MILLIS);
           appcoinsBilling = new UriCommunicationAppcoinsBilling(messageRequester);
         } else {
           appcoinsBilling = AppcoinsBilling.Stub.asInterface(service);
         }
-        return new AppcoinsBillingWrapper(appcoinsBilling,
-            sharedPreferencesRepository.getWalletId(), MESSAGE_RESPONSE_WAIT_TIMEOUT_IN_MILLIS);
+        return new AppcoinsBillingWrapper(appcoinsBilling, AppCoinsPendingIntentCaller.instance,
```

Comment:
if AppCoinsPendingIntentCaller.instance is a singleton why pass through a constructor parameter